fitbenchmarking/fitting_benchmarking.py

```diff
@@ -61,13 +61,43 @@ def do_fitting_benchmark(group_name, software_options, data_dir,
   user_input = misc.save_user_input(software, minimizers, group_name,
                                     group_results_dir, use_errors)
 
-  prob_results = None
-  prob_results = \
-      [do_fitbm_group(user_input, block) for block in problem_groups[group_name]]
+  prob_results = do_benchmarking(user_input, problem_groups, group_name)
 
   return prob_results, results_dir
 
 
+def do_benchmarking(user_input, problem_groups, group_name):
+  """
+  Loops through software and benchmarks each problem within the problem
+  group.
+
+  @param user_input :: all the information specified by the user
```

Comment:
@wathen your comment for user_input is correct. However, separately to this PR we could discuss how user_input may benefit from being broken down, i.e. to reduce the need for the code reader to have to look elsewhere in the code to check what user_input might contain